node/impl/full.go

```diff
@@ -117,4 +123,82 @@ func (n *FullNodeAPI) NodeStatus(ctx context.Context, inclChainStatus bool) (sta
 	return status, nil
 }
 
+// This method had been kept separate from wallet group implementation as it is not required for basic lotus functions
+// This method is exclusively used for Non Interactive Authentication based on worker address
+func (n *FullNodeAPI) FilIdSp(ctx context.Context, maddr address.Address) ([]byte, error) {
+	head, err := n.ChainHead(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get chain head: %w", err)
+	}
+
+	finality := abi.ChainEpoch(uint64(head.Height()) - 900)
+	ftipset, err := n.ChainGetTipSetByHeight(ctx, finality, types.EmptyTSK)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get finality tipset: %w", err)
+	}
+	beacon, err := n.StateGetBeaconEntry(ctx, head.Height())
```

Comment:
I would move this above the finality reach, it makes it easier to follow what happens now vs what happens in the past, instead of jumping back and forth.